api/base/content_negotiation.py

```diff
@@ -0,0 +1,38 @@
+from rest_framework import exceptions
+from rest_framework.utils.mediatypes import _MediaType
+from rest_framework.negotiation import DefaultContentNegotiation
+from rest_framework.utils.mediatypes import order_by_precedence, media_type_matches
+
+
+class CustomClientContentNegotiation(DefaultContentNegotiation):
```

Comment:
Can you explain why this class is necessary? Except for L23-24, `select_renderer` appears to be identical to the parent class's implementation, which is not ideal.